solana/solana-ibc/programs/solana-ibc/src/lib.rs

```diff
@@ -229,16 +234,18 @@ pub struct PrivateStorage {
 
 /// All the structs from IBC are stored as String since they dont implement AnchorSerialize and AnchorDeserialize
 #[derive(Debug)]
-pub struct IbcStorageInner<'a, 'b> {
+pub struct IbcStorageInner<'a, 'b, 'c, 'd> {
     pub private: &'a mut PrivateStorage,
     pub provable:
         solana_trie::AccountTrie<core::cell::RefMut<'a, &'b mut [u8]>>,
+    pub accounts: Vec<AccountInfo<'c>>,
+    pub signer: AccountInfo<'d>,
 }
```

Comment:
I’m fairly certain you can use single lifetime for `'c` and `'d`.  In fact you may be able to use `'b` for those lifetimes.

Reply:
We couldn't use same lifetimes since remaining accounts and signer from account struct have different lifetimes. But we are not using signer anyways, so I will remove it. I tried using 'b but it didn't work